Dashboard content.component

Arthur-Gev2023 opened the pull request.

Arthur-Gev2023 committed:
intialize request-component

Arthur-Gev2023 committed:
RequestComponent Created

-Delete an import in filterDashboard
- Create a newFile dasboard-requets-component
- Update a new css for testing the component
-Create fetch api to request api on request service
- Delete a useless fetch on project service

robinjessondecathlon commented on src/app/component/dashboard-request.component/dasboard-request-content.component.tsx:
tu peux simplifier ce composant : 
- tu dois avoir un callApiToGetRequestById(id: number)
- tu n'as qu'un seul useEffect : 
```ts
useEffect(() => { 
   /* le but de ce useEffect, c'est qu'à chaque fois que props.requestId change, il faut que tu appelles le service http /api/request/:id */
}, [props.requestId]); 
```

Arthur-Gev2023 committed:
affichage de la requête sélectionnée

-updated dashboard request content component
-change request service all to byId
-change on dashboard filter component to have the good request id on start

Arthur-Gev2023 committed:
Affichage des Requêtes

-Updated Dashboard-filter-component
-Updated Dasboard-Request-Content-Component
- Change on page.tsx, delete condition

Arthur-Gev2023 committed:
nettoyage du front

- supprime les console log
- commente le nv code
- ajoute les ; en bout de ligne

robinjessondecathlon commented on src/app/component/dashboard-request.component/dasboard-request-content.component.tsx:
```diff
@@ -0,0 +1,66 @@
+  return (
+    <Fragment>
+      {
+        props.requestId === undefined || !selectedRequest ? (
```
le if + retour était plus clair 
et sans le Fragment

robinjessondecathlon commented on src/app/component/dashboard-filter.component/dashboard-filter.component.tsx:
```diff
@@ -0,0 +1,157 @@
+          },
+          // Callback exécuté lorsque l'Observable est terminé
+          complete: () => {
+            console.log('API call completed successfully');
```
supprime ce complete, on en a pas besoin

robinjessondecathlon commented on src/app/component/dashboard-filter.component/dashboard-filter.component.tsx:
```diff
@@ -0,0 +1,157 @@
+        // Retour d'une fonction de nettoyage pour annuler la souscription à l'Observable
+        return () => subscription.unsubscribe();
+      } catch (error) {
+        console.log(error);
```
supprime le console log ici

Arthur-Gev2023 committed:
Change on Front

- Change on dashboard-filter-component ( Delete console.log , delete complete)
-Change on RequestContent-component
if + return
